\hypertarget{classsocketpp_1_1SockException}{
\section{socketpp::SockException Class Reference}
\label{classsocketpp_1_1SockException}\index{socketpp::SockException@{socketpp::SockException}}
}
socket++ exception class  


{\tt \#include $<$SockException.h$>$}

Inherits std::exception.

\subsection*{Public Member Functions}
\begin{CompactItemize}
\item 
\hyperlink{classsocketpp_1_1SockException_ea2b065bb1b5c6bf062b6a9d7574853f}{SockException} (const std::string \&meth, const std::string \&err, const std::string \&func=\char`\"{}\char`\"{})  throw ()
\item 
\hyperlink{classsocketpp_1_1SockException_b83e8ced6e6fa142711d0a49e3e70ba0}{SockException} (const std::string \&meth, int err, const std::string \&func=\char`\"{}\char`\"{})  throw ()
\item 
virtual const char $\ast$ \hyperlink{classsocketpp_1_1SockException_5c15a1c818818042230d017d924ef9ea}{what} () const   throw ()
\begin{CompactList}\small\item\em returns complete error string \item\end{CompactList}\item 
int \hyperlink{classsocketpp_1_1SockException_cc8b24550b7a77c5f710a6abaff6e696}{getErrno} ()
\begin{CompactList}\small\item\em returns errno code \item\end{CompactList}\item 
\hyperlink{classsocketpp_1_1SockException_5ab9299ca51c12d5d1a62d7872003b84}{$\sim$SockException} ()  throw ()
\end{CompactItemize}


\subsection{Detailed Description}
socket++ exception class 

\subsection{Constructor \& Destructor Documentation}
\hypertarget{classsocketpp_1_1SockException_ea2b065bb1b5c6bf062b6a9d7574853f}{
\index{socketpp::SockException@{socketpp::SockException}!SockException@{SockException}}
\index{SockException@{SockException}!socketpp::SockException@{socketpp::SockException}}
\subsubsection[{SockException}]{\setlength{\rightskip}{0pt plus 5cm}socketpp::SockException::SockException (const std::string \& {\em meth}, \/  const std::string \& {\em err}, \/  const std::string \& {\em func} = {\tt \char`\"{}\char`\"{}})  throw ()\hspace{0.3cm}{\tt  \mbox{[}inline\mbox{]}}}}
\label{classsocketpp_1_1SockException_ea2b065bb1b5c6bf062b6a9d7574853f}


\begin{Desc}
\item[Parameters:]
\begin{description}
\item[{\em meth}]method which threw exception \item[{\em err}]error description string \item[{\em func}]C function which returned error \end{description}
\end{Desc}
\hypertarget{classsocketpp_1_1SockException_b83e8ced6e6fa142711d0a49e3e70ba0}{
\index{socketpp::SockException@{socketpp::SockException}!SockException@{SockException}}
\index{SockException@{SockException}!socketpp::SockException@{socketpp::SockException}}
\subsubsection[{SockException}]{\setlength{\rightskip}{0pt plus 5cm}socketpp::SockException::SockException (const std::string \& {\em meth}, \/  int {\em err}, \/  const std::string \& {\em func} = {\tt \char`\"{}\char`\"{}})  throw ()\hspace{0.3cm}{\tt  \mbox{[}inline\mbox{]}}}}
\label{classsocketpp_1_1SockException_b83e8ced6e6fa142711d0a49e3e70ba0}


\begin{Desc}
\item[Parameters:]
\begin{description}
\item[{\em meth}]method which threw exception \item[{\em err}]errno code \item[{\em func}]C function which returned error \end{description}
\end{Desc}
\hypertarget{classsocketpp_1_1SockException_5ab9299ca51c12d5d1a62d7872003b84}{
\index{socketpp::SockException@{socketpp::SockException}!$\sim$SockException@{$\sim$SockException}}
\index{$\sim$SockException@{$\sim$SockException}!socketpp::SockException@{socketpp::SockException}}
\subsubsection[{$\sim$SockException}]{\setlength{\rightskip}{0pt plus 5cm}socketpp::SockException::$\sim$SockException ()  throw ()\hspace{0.3cm}{\tt  \mbox{[}inline\mbox{]}}}}
\label{classsocketpp_1_1SockException_5ab9299ca51c12d5d1a62d7872003b84}




\subsection{Member Function Documentation}
\hypertarget{classsocketpp_1_1SockException_cc8b24550b7a77c5f710a6abaff6e696}{
\index{socketpp::SockException@{socketpp::SockException}!getErrno@{getErrno}}
\index{getErrno@{getErrno}!socketpp::SockException@{socketpp::SockException}}
\subsubsection[{getErrno}]{\setlength{\rightskip}{0pt plus 5cm}int socketpp::SockException::getErrno ()\hspace{0.3cm}{\tt  \mbox{[}inline\mbox{]}}}}
\label{classsocketpp_1_1SockException_cc8b24550b7a77c5f710a6abaff6e696}


returns errno code 

\hypertarget{classsocketpp_1_1SockException_5c15a1c818818042230d017d924ef9ea}{
\index{socketpp::SockException@{socketpp::SockException}!what@{what}}
\index{what@{what}!socketpp::SockException@{socketpp::SockException}}
\subsubsection[{what}]{\setlength{\rightskip}{0pt plus 5cm}virtual const char$\ast$ socketpp::SockException::what () const  throw ()\hspace{0.3cm}{\tt  \mbox{[}inline, virtual\mbox{]}}}}
\label{classsocketpp_1_1SockException_5c15a1c818818042230d017d924ef9ea}


returns complete error string 



The documentation for this class was generated from the following file:\begin{CompactItemize}
\item 
\hyperlink{SockException_8h}{SockException.h}\end{CompactItemize}
